sktime/transformations/series/adi_cv.py

```diff
@@ -153,51 +178,61 @@ def _transform(self, X, y=None):
             during initialization. Specifically, the columns include (by default):
 
                 1. Average Demand Interval (ADI)
-                2. Variance (CV2)
+                2. Coefficient of Variation squared (CV2)
                 3. categorical class
         """
-        X_non_zero = X.to_numpy().nonzero()
-        X_non_zero = X.iloc[X_non_zero]
+        X_non_zero_indices = X.to_numpy().nonzero()[0]
+        X_non_zero_values = X.iloc[X_non_zero_indices].values
 
-        # Calculating ADI value based on formula from paper
-        adi_value = len(X) / (len(X_non_zero) - 1)
+        # Calculating ADI value based on adi_trim_handling parameter
+        n_non_zero_observations = len(X_non_zero_indices)
+        n_observations = len(X)
 
-        # Calculating variance for all non-zero values
-        variance = X_non_zero.var().iloc[0]
-        cv2_value = variance / len(X_non_zero)
+        if self.adi_trim_handling == "trim":
+            first_non_zero = X_non_zero_indices[0]
+            last_non_zero = X_non_zero_indices[-1]
 
-        # Calculating the class type
+            adi_numerator = last_non_zero - first_non_zero
+            adi_denominator = n_non_zero_observations - 1
 
-        adi_low = adi_value <= self.adi_threshold
-        cv2_low = cv2_value <= self.cv_threshold
-        adi_high = not adi_low
-        cv2_high = not cv2_low
+        elif self.adi_trim_handling == "pool":
+            adi_numerator = n_observations
+            adi_denominator = n_non_zero_observations
 
-        if adi_low and cv2_low:
-            class_type = "smooth"
+        elif self.adi_trim_handling == "ignore":
+            adi_numerator = n_observations
+            adi_denominator = n_non_zero_observations - 1
 
-        elif adi_low and cv2_high:
-            class_type = "erratic"
+        adi_value = adi_numerator / adi_denominator
 
-        elif adi_high and cv2_low:
-            class_type = "intermittent"
+        # Calculating coefficient of variation squared for all non-zero values
+        sigma = np.std(X_non_zero_values)
+        mu = np.mean(X_non_zero_values)
+        cv2_value = (sigma / mu) ** 2
 
-        elif adi_high and cv2_high:
-            class_type = "lumpy"
+        # Calculating the class type
+        if adi_value < self.adi_threshold:
+            if cv2_value < self.cv_threshold:
+                class_type = "smooth"
+            elif cv2_value >= self.cv_threshold:
+                class_type = "erratic"
+        elif adi_value >= self.adi_threshold:
+            if cv2_value < self.cv_threshold:
+                class_type = "intermittent"
+            elif cv2_value >= self.cv_threshold:
+                class_type = "lumpy"
 
         # Collecting all values together into dict and converting to DF
-        return_dict = {}
```

Comment:
I think the original idea of invoking the `pd.DataFrame` constructor only one is more robust. Assignments in `DataFrame` via the brackets can be buggy, imo we should avoid them. I could tell stories about these breaking after `pandas` updates because the coercion logic changes.